fuse_models/package.xml

```diff
@@ -13,6 +13,22 @@
 
   <build_depend>roslint</build_depend>
 
+  <!-- The official rosdep doesn't have an entry for Google's benchmark deb:
+       https://github.com/ros/rosdistro/blob/master/rosdep/base.yaml
+
+       There's an official deb named libbenchmark-dev available since Ubuntu Bionic:
+       https://packages.ubuntu.com/bionic/libbenchmark-dev
+
+       ROS Melodic officially supports Ubuntu Bionic:
+       https://www.ros.org/reps/rep-0003.html
+
+       This could be uncommented if we add a benchmark entry like this to the official rosdep:
+
+       benchmark:
+         ubuntu:
+           bionic: libbenchmark-dev
+  -->
+  <!--<depend>benchmark</depend>-->
```

Comment:
Should this be a `test_depend`?

Reply:
Good point. IMHO it's good to have the ability to run the benchmark on the robot platforms, so in that sense it's better to have the benchmark executable available not only in a test build. Note that in the `CMakeLists.txt` I also added an `install` rule for it.

Either way, this is behind another cmake flag, so it actually doesn't compile by default and it wouldn't be available. `-DCATKIN_ENABLE_BENCHMARKING=1` must be passed to `cmake` or `catkin build --cmake-args` for that.

That being said, I don't have a strong opinion, so I can change it if you think there's no point on having a benchmark generally available.

Reply:
> Either way, this is behind another cmake flag, so it actually doesn't compile by default and it wouldn't be available. `-DCATKIN_ENABLE_BENCHMARKING=1` must be passed to `cmake` or `catkin build --cmake-args` for that.

true, but the dependency is installed regardless.

Reply:
> IMHO it's good to have the ability to run the benchmark on the robot platforms, so in that sense it's better to have the benchmark executable available not only in a test build.

So it sounds like this isn't being built from source on the robot platform in question, or you could just put the new `CATKIN_ENABLE_BENCHMARKING` block inside the `CATKIN_ENABLE_TESTING` block, I think.

Would expanding the rosdep platforms for the dependency address your concern, @gavanderhoorn?

Reply:
> you could just put the new `CATKIN_ENABLE_BENCHMARKING` block inside the `CATKIN_ENABLE_TESTING` block

That would indeed perhaps be the best thing to do.

> Would expanding the rosdep platforms for the dependency address your concern,

As we don't have optional dependencies in ROS package manifests, anything you add will become a required dependency. I'm not too worried about this one in particular, but it will make setting up a build environment for `fuse` a little more complex.

Reply:
I see your concerns. I think in brief you're asking to keep things simply, so what about:

1. Change `<depend>` into `<test_depend>` for `benchmark`
2. Remove the `CATKIN_ENABLE_BENCHMARKING` block and move the benchmark inside `CATKIN_ENABLE_TESTING`. It won't run, but it'll get build in that case.
3. With the benchmark inside the `CATKIN_ENABLE_TESTING` block, I can remove the `install` rule for the benchmark. I should be able to find where the executable ends up, so there must be a way to call it.

This means that in order to run the benchmark on a robot you'll have to:

1. Create a workspace with `fuse`
2. Run `catkin build` in order to get the `build/fuse_models` folder. Not sure if there's a faster way.
3. Go into the `build/fuse_models` folder and from there there should be a `make` target for the benchmark, that we can build alone
4. Finally, we should be able to run the benchmark

If you agree on this, I can make the changes to unblock this PR. :smiley:

Reply:
Works for me. The executable should live in: `{catkin_workspace}/devel/lib/fuse_models`
At least if you are using `catkin build`.

Reply:
Updated. Now:
1. There's a `<test_depend>` on `benchmark` as opposed to a `<depend>`
2. I updated the comment in the `package.xml` with the link to the existing PR in the official rosdistro, which has been approved but not merged yet.

- I'll remove that entire comment and uncomment the `<test_depend>` when that gets merged, although it also means you need to sync the rosdistro in your buildfarm for it to work.

- This will make the benchmark compile, but it won't run as a test. I guess we could consider that this way we are at least building the benchmark, i.e. testing it builds, which is better than nothing.

3. I removed the `CATKIN_ENABLE_BENCHMARKING` block and moved the benchmark inside (at the botom) the `CATKIN_ENABLE_TESTING` block.

- However, `catkin build` seems to default to `CATKIN_ENABLE_TESTING=ON`, so it tries to build the benchmark. Note that because it's not a `gtest`, the benchmark target is built immediately. There's no need for `make run_tests`. For this reason, if the `benchmark` package is NOT available in the system it'd fail to build.

4. To solve that issue, I made the `find_package` on `benchmark` not `REQUIRED` (now is `QUIET`), and the benchmark is built only if `benchmark_FOUND` is set to `TRUE`.

I tested this by removing the pkgconfig and cmake benchmark stuff from my system, so the `benchmark` package could NOT be found. It worked:

- No cmake errors/warnings because `benchmark` wasn't found, thanks to the `QUIET` mode in `find_package`
- The benchmark wasn't built

Then I brought the removed files back, touched the `CMakeLists.txt` file to force a re-build and `catkin build` again. After that, the benchmark was built. The resulting executable goes here (one is a symlink):
``` bash
~/ws/devel$ find | grep benchmark
./lib/fuse_models/benchmark_unicycle_2d_state_cost_function
./.private/fuse_models/lib/fuse_models/benchmark_unicycle_2d_state_cost_function
```

And it can be run with `rosrun fuse_models benchmark_unicycle_2d_state_cost_function`.